dex/src/main/scala/com/wavesplatform/dex/queue/KafkaMatcherQueue.scala

```diff
@@ -110,17 +111,17 @@ class KafkaMatcherQueue(settings: Settings) extends MatcherQueue with ScorexLogg
       }
   }
 
-  override def close(timeout: FiniteDuration): Unit = {
+  override def close(timeout: FiniteDuration): Future[Unit] = {
     duringShutdown.set(true)
-
-    producer.close(timeout)
-    producerThreadPool.shutdown()
-
-    // Fix by scala.jdk.DurationConverters after migration to Scala 2.13
-    val duration = java.time.Duration.ofNanos(timeout.toNanos)
-    Await.ready(Future(consumer.close(duration))(consumerExecutionContext), timeout)
-    consumerTask.cancel()
-    consumerThreadPool.shutdown()
+    Future {
+      blocking {
+        producer.close(timeout)
+        producerThreadPool.shutdown()
+        consumer.close(timeout.toJava)
+        consumerTask.cancel()
+        consumerThreadPool.shutdown()
+      }
+    }(scala.concurrent.ExecutionContext.global)
```

Comment:
Why global ec?

Reply:
Because we can't run a Future in `consumerThreadPool`, it closed in the Future :)